packages/ffe-cards-react/src/NewCard.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import classNames from 'classnames';
+import { oneOfType, node, func, string, bool } from 'prop-types';
+
+const NewCard = props => {
+    const {
+        className,
+        details,
+        element: Element,
+        heading,
+        icon,
+        condensed,
+        greyCharcoal,
+        ...rest
+    } = props;
+
+    return (
+        <Element
+            className={ classNames(
+                'ffe-new-card',
+                { 'ffe-new-card--condensed': condensed },
+                { 'ffe-new-card--grey-charcoal': greyCharcoal },
+                className
+            ) }
+            { ...rest }
+        >
+            {icon &&
+            React.cloneElement(icon, {
+                className: classNames(
+                    'ffe-new-card__icon',
+                    icon.props.className
+                ),
+            })}
+            <div className="ffe-new-card__body">
+                {heading && <p className="ffe-new-card__heading">{heading}</p>}
+                {details && <p className="ffe-new-card__details">{details}</p>}
```

Comment:
A component doesn't need to have children - in cases like these, they typically don't (since things need to be in a certain order.

An alternative could be to use render props with what's known as compound components:

```js
<NewCard>
  {({ Heading, Details }) => (
    <Fragment>
      <Heading element="h2">My heading</Heading>
      <Details>This is some details</Details>
    </Fragment>
  )}
</NewCard>
```

, but I think that's major overkill.

Strict APIs that makes sure the consumer doesn't abuse a component's intended usecase might even lead to better DX (developer experience) in the long term, because we know what to use what component for.

Reply:
Hard to answer that without knowing how this component will be used in 1 year's time ;)

Reply:
That's a cool technique @selbekk ! I never considered you could return components for render props.
That's true @ivarni . I just hope we can agree on some way to do it now. Nothing stands in the way of breaking that in a year's time and doing these cards in a different way.